ote/src/clj/ote/transit_changes/detection.clj

```diff
@@ -946,6 +930,82 @@
   [route-list-with-week-hashes]
   (vec (mapcat route-differences route-list-with-week-hashes)))
 
+(defn- route-ends? [^LocalDate date max-date ^Integer traffic-threshold-d]
```

Comment:
If you have extra energy you could consider moving all this change type detection related codes to different file.
This add-ending-route-change fn contains multiple inner functions and in a different file it would be possible to separate them.
Also, it would be easier to create unit tests if functions were smaller.